lib/scope.js

```diff
@@ -116,30 +116,38 @@ ScopeDefinition.prototype.related = function(receiver, scopeParams, condOrRefres
         var relatedModel = targetModel.relations[filter.relation].modelTo;
         var IdKey = idName(relatedModel);
 
-        var smartMerge = function(idCond, qWhere, idKey) {
+        // return {inq: [1,2,3]}}
+        var smartMerge = function(idCollection, qWhere) {
+          if (!qWhere[IdKey]) return idCollection;
           var merged = {};
-          var idsA = idCond[idKey].inq;
-          var idsB = qWhere[idKey].inq ? qWhere[idKey].inq : [qWhere[idKey]];
+
+          var idsA = idCollection.inq;
+          var idsB = qWhere[IdKey].inq ? qWhere[IdKey].inq : [qWhere[IdKey]];
```

Comment:
@joostdebruijn hmm, a question, compared with my previous fix, is there any reason why taking out the validation for `typeof inq`?

```js
if (!Array.isArray(qWhere[IdKey].inq)) {
           return qWhere[IdKey];
        }
```

Reply:
this is the code I am talking about:
https://github.com/strongloop/loopback-datasource-juggler/pull/1491/files#diff-c2a679b29102d5574291e884531cb8aeR126

Reply:
Yes, the check caused that in some cases the lines [128/129](https://github.com/joostdebruijn/loopback-datasource-juggler/blob/dc20fe1d8e809e46007cf1f4f233d6fe3e095213/lib/scope.js#L128) where not executed because of the returing, this led to unexpected query results. When I removed the check, the results were as expected. Maybe we can add a test case which validates if the case you were validating is handled correctly?

Reply:
@joostdebruijn If user provide an invalid value in `inq`, e.g. `{where: {id: {inq: 'INVALID'}}}`, my code will pass it into the query validator which will returns error, but taking it out results in the merge fails and passes `{}` as id query.....

Reply:
@joostdebruijn Anyway don't worry I will add a test case and continue the fix in my PR :) Thank you for contribution!! :ship:

Reply:
@jannyHou Ah, I understand the problem! Let me know if I can help you in some way or review the next PR or testcase.